src/internal/clusterstate/v2.8.0/pfsdb.go

```diff
@@ -295,8 +295,8 @@ func alterCommitsTablePostDataMigration(ctx context.Context, env migrations.Env)
 func createCommitAncestryTable(ctx context.Context, tx *pachsql.Tx) error {
 	query := `
 	CREATE TABLE pfs.commit_ancestry (
-		parent bigint REFERENCES pfs.commits(int_id),
-		child bigint REFERENCES pfs.commits(int_id),
+		parent bigint REFERENCES pfs.commits(int_id) ON DELETE CASCADE,
+		child bigint REFERENCES pfs.commits(int_id) ON DELETE CASCADE,
```

Comment:
What is the intention here, when we delete a commit will it also wipe out it's ancestors and descendants?

Reply:
it will remove ancestry information for that commit. We still delete the ancestry information ourselves (i.e. repoint the parent and children) before we do a commit delete. I think this is just to safeguard us in a situation where a commit is somehow deleted outside of our API.

Reply:
Ohh this is the ancestry table.  So if we delete from the commits table, this would be a dangling reference.  Makes sense 👍 .